src/lib.rs

```diff
@@ -1238,6 +1238,60 @@ impl InterruptHandle {
     }
 }
 
+/// Adds an entrypoint implementation for a SQLite extension.
+/// This macro must be used once in the root of a crate that implements a SQLite extension.
+///
+/// # Example
+/// ```
+/// use rusqlite::Connection;
+/// use rusqlite::ffi::SQLITE_NOTICE;
+/// use rusqlite::sqlite3_extension_init;
+/// use rusqlite::trace::log;
+///
+/// sqlite3_extension_init!(init_loadable_extension);
+///
+/// fn init_loadable_extension(db: Connection) -> ::rusqlite::Result<()> {
+///    db.create_scalar_function(
+///         "rusqlite_test_function",
+///         0,
+///         FunctionFlags::SQLITE_DETERMINISTIC,
+///         |_ctx| {
+///             Ok(ToSqlOutput::Owned(Value::Text(
+///                 "Rusqlite extension loaded correctly!".to_string(),
+///             )))
+///         },
+///     )?;
+///     log(SQLITE_NOTICE, "Rusqlite extension initialized");
+///     Ok(())
+/// }
+///```
+#[cfg(feature = "loadable_extension")]
+#[cfg_attr(docsrs, doc(cfg(feature = "loadable_extension")))]
+#[macro_export]
+macro_rules! sqlite3_extension_init {
+    ($func: ident) => {
+        #[allow(clippy::not_unsafe_ptr_arg_deref)]
+        #[no_mangle]
+        pub extern "C" fn sqlite3_extension_init(
```

Comment:
https://sqlite.org/loadext.html#programming_loadable_extensions
> You will do well to customize the name of your entry point to correspond to the name of the shared library you will be generating, rather than using the generic "sqlite3_extension_init" name. Giving your extension a custom entry point name will enable you to statically link two or more extensions into the same program without a linker conflict, if you later decide to use static linking rather than run-time linking.

Reply:
@gwenn thx, good point, thx. Looking further at [sqlite3_load_extension](https://sqlite.org/c3ref/load_extension.html), unless the user actually calls an API or specifies entry point by hand, sqlite uses heuristic to find this entry point automatically:

> t will first try the generic extension name "sqlite3_extension_init". If that does not work, it constructs a entry point using the template "sqlite3_X_init" where the X is replaced by the lowercase equivalent of every ASCII character in the filename after the last "/" and before the first following "." omitting the first three characters if they happen to be "lib". So, for example, if the filename is "/usr/lib/libmathfunc-4.8.so" the entry point name would be "sqlite3_mathfunc_init". Or if the filename is "./SpellFixExt.dll" then the entry point would be called "sqlite3_spellfixext_init".

Should the macro take optional parameter to override the extension name (in the spirit of lower entry barrier?).  Otherwise each user would have to read this doc and decide the name of their extension.